File: RuntimeData/runtime/app/src/main/java/Actions/ACT_STOPLOOP.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
// -----------------------------------------------------------------------------
//
// STOP LOOP
//
// -----------------------------------------------------------------------------
package Actions;

import Params.CParamExpression;
import RunLoop.CRun;

public class ACT_STOPLOOP extends CAct
{
    public void execute(CRun rhPtr)
    {
        CLoop pLoop = rhPtr.findFastLoop
                (rhPtr.get_EventExpressionStringLowercase((CParamExpression)evtParams[0]));

        if (pLoop == null)
            return;

        pLoop.flags|=CLoop.FLFLAG_STOP;
    }
}

File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Actions/ACT_STOPLOOP.cs
// -----------------------------------------------------------------------------
//
// STOP LOOP
//
// -----------------------------------------------------------------------------
using System;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Params;
namespace RuntimeXNA.Actions
{
	
	public class ACT_STOPLOOP:CAct
	{
		public override void  execute(CRun rhPtr)
		{
			System.String name = rhPtr.get_EventExpressionString((CParamExpression) evtParams[0]);
			if (name.Length == 0)
				return ;
			
			CLoop pLoop;
			int n;
			for (n = 0; n < rhPtr.rh4FastLoops.size(); n++)
			{
				pLoop = (CLoop)rhPtr.rh4FastLoops.get(n);
				if (System.String.Compare(pLoop.name, name, StringComparison.OrdinalIgnoreCase) == 0)
				{
					pLoop.flags |= CLoop.FLFLAG_STOP;
				}
			}
		}
	}
}